activities/ActivityTemplateVue.activity/js/components/journal.js

```diff
@@ -0,0 +1,54 @@
+var Journal = {
+  data: {
+		parent: null,
+	},
+  mounted() {
+		this.parent = this.$parent;
+    
+    let vm = this;
+    requirejs(["sugar-web/activity/activity", "sugar-web/env"], function (activity, env) {
+			env.getEnvironment(function (err, environment) {
+				
+				// Load context
+				if (environment.objectId) {
+					console.log("Existing instance");
+					activity.getDatastoreObject().loadAsText(function (error, metadata, data) {
```

Comment:
I wonder if all callback should not be implemented as event

Reply:
Emits an event on data load now.